flock_agent/daemon/flock_logs.py

```diff
@@ -0,0 +1,140 @@
+import os
+import time
+import json
+
+from .api_client import FlockApiClient
+
+
+class FlockLog:
+    def __init__(self, common, lib_dir):
+        self.c = common
+        self.filename = os.path.join(lib_dir, "flock.log")
+        self.c.log("FlockLog", "__init__", self.filename)
+
+        # If the log file doesn't exist, create an empty one
+        if not os.path.exists(self.filename):
+            open(self.filename, "a").close()
+            os.chmod(self.filename, 0o600)
+
+    def log(self, flock_log_type, twig_ids=None):
+        with open(self.filename, "a") as f:
```

Comment:
> I don't think file corruption is an issue because of how asyncio works -- it runs everything in a single process, in an event loop, so it's more similar to like async in javascript than to multiple threads. Only one line of code ever runs at a time, so there should never be a situation where the log is getting written to twice simultaneously.